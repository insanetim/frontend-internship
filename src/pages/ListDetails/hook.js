import { useState, useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useNavigate, useParams } from 'react-router-dom'
import { Modal } from 'antd'

import { deleteList, removeFromList, fetchList } from 'src/state/lists/actions'
import { listSelector } from 'src/state/lists/selectors'

const useContainer = () => {
  const dispatch = useDispatch()
  const list = useSelector(listSelector)
  const { listId } = useParams()
  const navigate = useNavigate()
  const [loading, setLoading] = useState(true)

  const handleListDelete = () => {
    Modal.confirm({
      title: 'Do you want to delete list?',
      onOk() {
        dispatch(deleteList(listId, () => navigate('/lists')))
      }
    })
  }

  const handleMovieDelete = (event, movieId) => {
    event.stopPropagation()
    Modal.confirm({
      title: 'Do you want to delete movie from this list?',
      onOk() {
        dispatch(removeFromList({ listId, movieId }))
      }
    })
  }

  useEffect(() => {
    dispatch(fetchList(listId, () => setLoading(false)))
  }, [listId])

  return { list, loading, handleListDelete, handleMovieDelete }
}

export default useContainer
