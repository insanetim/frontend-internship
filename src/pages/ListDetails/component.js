import React from 'react'
import { Row, Col, Typography } from 'antd'
import { MinusCircleOutlined, DeleteOutlined } from '@ant-design/icons'

import Loading from 'src/components/Loading'
import ListContent from 'src/components/ListContent'
import { useContainer } from './hook'

const ListDetails = () => {
  const { list, loading, handleListDelete, handleMovieDelete } = useContainer()

  if (loading) {
    return (
      <div className='top-margin'>
        <Loading />
      </div>
    )
  }

  return (
    <>
      <Row>
        <Col
          offset={2}
          span={20}
        >
          <div className='top-margin'>
            <Typography.Title>
              {list.name} <MinusCircleOutlined onClick={handleListDelete} />
            </Typography.Title>
          </div>
        </Col>
      </Row>
      <ListContent
        movies={list.items}
        actions={[
          <DeleteOutlined
            key='delete'
            onClick={handleMovieDelete}
          />
        ]}
      />
    </>
  )
}

export default ListDetails
