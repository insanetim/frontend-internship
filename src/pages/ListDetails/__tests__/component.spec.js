import { shallow } from 'enzyme'

import ListDetails from '../component'

const mockedHookData = {
  list: {
    name: 'Test list',
    items: []
  },
  loading: false,
  handleListDelete: jest.fn(),
  handleMovieDelete: jest.fn()
}
jest.mock('../hook', () => jest.fn(() => mockedHookData))

it('matches snapshot', () => {
  const component = shallow(<ListDetails />)

  expect(component).toMatchSnapshot()
})

it('matches snapshot with loading', () => {
  mockedHookData.loading = true
  const component = shallow(<ListDetails />)

  expect(component).toMatchSnapshot()
})
