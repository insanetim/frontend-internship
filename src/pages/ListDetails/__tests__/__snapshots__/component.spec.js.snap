// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <Row>
    <Col
      offset={2}
      span={20}
    >
      <div
        className="top-margin"
      >
        <ForwardRef>
          Test list
           
          <ForwardRef(MinusCircleOutlined)
            onClick={[MockFunction]}
          />
        </ForwardRef>
      </div>
    </Col>
  </Row>
  <ListContent
    actions={
      [
        <ForwardRef(DeleteOutlined)
          onClick={[MockFunction]}
        />,
      ]
    }
    movies={[]}
  />
</Fragment>
`;

exports[`matches snapshot with loading 1`] = `
<div
  className="top-margin"
>
  <Loading />
</div>
`;
